lib/managers/window_manager.cpp

```diff
@@ -98,3 +99,67 @@ void WindowManager::update_title() {
 
     glfwSetWindowTitle(window_, stream.str().c_str());
 }
+
+void WindowManager::refresh() {
+    if (!valid()) return;
+
+    double time = WorldTimer::get_time_sec();
+
+    if (last_title_update_ + subtitle_update_dt_ < time) {
+        update_title();
+    }
+
+    glfwSwapBuffers(get_active_window());
+}
+
+WindowManager::SubtitleEntry WindowManager::add_subtitle_entry(
+    const std::string& key, double frequency) {
+    double delta_time = frequency == 0.0 ? 0.0 : 1.0 / frequency;
```

Comment:
I think it's odd to see different behavior when the frequency is zero or some other number very close to zero. Perhaps we should use time instead of frequency and avoid making any divisions.

Reply:
Fixed